File: nodescript/hosts/bayfiles.js
const path = require('path')
const fs = require('fs')
const {exec, execSync} = require('child_process')



module.exports = function(files = []) {

  return new Promise((resolve, reject) => {

    let results = []


    for(const f of files) {
      let command = `curl -s \
      https://api.bayfiles.com/upload?token=3920895f095f5e82 \
      -F 'file=@${f}'`

      //let command = `curl -F file=@/mnt/e/nodeprojects/serverhost/storage/1922mp4/capture-video.png https://srv-file6.gofile.io/uploadFile`
      command  = execSync(command)
      try {
        let output = JSON.parse(command.toString('utf8'))
        console.log(output.data.file.url.full)
        results.push(output.data.file.url.full)
      } catch (error) { console.log(error)}

    }

    resolve(results)


  })

}

File: python/api/bayfiles.py
from os import path
import json
import requests
import re

# comment line


def do_upload(pathfile):
  #pathfile = path.join(path.dirname(path.abspath(__file__)), 'video2.mp4')
  if(path.isfile(pathfile) == False):
    return {"error": "file no exists"}
  
  try:
    f = open (pathfile,'rb')
    r = requests.post(
      url="https://api.bayfiles.com/upload?token=3920895f095f5e82", 
      files={'file':f}
    )
   
    #print (r.content)
    js = json.loads(r.content)

    return js['data']['file']['url']['full']

  except:
    return {"error": "ocurrio un error con los archivos"}
      
 
    


